docker/apm-server/managed/docker-entrypoint.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -e
+
+echo ${KIBANA_HOST}
+
+fleet="${KIBANA_HOST}/api/fleet"
+pkg_policy_name="apm-integration-testing"
+
+# remove existing apm-integration-testing package policy
+pkg_policy_id="$(curl -s -H 'kbn-xsrf: true' -H 'Content-Type: application/json' "${fleet}/package_policies?kuery=ingest-package-policies.name:${pkg_policy_name}" | jq '.items[0].id')"
+if [ -n "$pkg_policy_id" ];then
+  pkg_policy_ids()
+{
+  cat <<EOF
+  {
+  "packagePolicyIds":[${pkg_policy_id}]
+  }
+EOF
+}
+  success="$(curl -s -H 'kbn-xsrf: true' -H 'Content-Type: application/json' "${fleet}/package_policies/delete" -d "$(pkg_policy_ids)" | jq '.[0].success')"
+  if [ "$success" = false ]; then
+    echo "deleting existing package policy failed"
+    exit -1
+  fi
+fi
+
+# fetch the default agent policy
+agent_policy_id="$(curl -s "${fleet}/agent_policies?kuery=ingest-agent-policies.is_default:true" | jq '.items[0] .id')"
+if [ -z $agent_policy_id ]; then
+    echo "no default agent policy found"
+    exit -2
+fi
+
+# create apm-integration-testing package policy
+# and aadd it to the default agent policy
+pkg_policy_data()
+{
+  cat <<EOF
+{
+  "name":"apm-integration-testing",
```

Comment:
@nchaulet I'm wondering how much of these json parts are optional and things would still work? Can we skip `output_id`, `namespace`, `streams` any many other things and the correct defaults will be filled in either by KB or Elastic Agent? This would make the API simpler.

Reply:
Right now all of these fields are mandatory in the API but I agree we can simplify the API and have optional field here